src/Platform/Microsoft.Testing.Platform/Configurations/AggregatedConfiguration.cs

```diff
@@ -61,7 +61,7 @@ public void SetTestHostWorkingDirectory(string workingDirectory) =>
     public async Task CheckTestResultsDirectoryOverrideAndCreateItAsync(ICommandLineOptions commandLineOptions, IFileLoggerProvider? fileLoggerProvider)
     {
         // Load Configuration
-        _currentWorkingDirectory = Path.GetDirectoryName(_testApplicationModuleInfo.GetCurrentTestApplicationFullPath())!;
+        _currentWorkingDirectory = _testApplicationModuleInfo.GetCurrentTestApplicationDirectory();
```

Comment:
Similar change as pointed out before. The implementation of `GetCurrentTestApplicationDirectory` is equivalent to the original code, except that we tolerate null and use `AppContext.BaseDirectory`.